File: lib/screens/login_screen.dart
import 'package:flutter/material.dart';
import 'package:google_fonts/google_fonts.dart';
import '../theme/colors.dart';

class LoginScreen extends StatefulWidget {
  final VoidCallback onLoginSuccess;

  const LoginScreen({super.key, required this.onLoginSuccess});

  @override
  State<LoginScreen> createState() => _LoginScreenState();
}

class _LoginScreenState extends State<LoginScreen> {
  final TextEditingController _usernameController = TextEditingController();
  final TextEditingController _passwordController = TextEditingController();
  bool _passwordVisible = false;
  bool _keepMeSignedIn = false;
  bool _isSimulatingLoading = false;
  String _errorMessage = '';

  Widget _buildLogo() {
    return Column(
      mainAxisSize: MainAxisSize.min,
      crossAxisAlignment: CrossAxisAlignment.center,
      children: [
        Text(
          "JLW",
          style: GoogleFonts.serif(
            fontWeight: FontWeight.black,
            fontSize: 42,
            color: SystemColors.corporateNavy,
            letterSpacing: 1,
            shadows: [
              Shadow(
                color: Colors.black.withOpacity(0.12),
                offset: const Offset(2, 2),
                blurRadius: 3,
              ),
            ],
          ),
        ),
        const SizedBox(height: 2),
        Text(
          "SINCE 1875",
          style: GoogleFonts.sansSerif(
            fontWeight: FontWeight.bold,
            fontSize: 10,
            color: SystemColors.textGray,
            letterSpacing: 4,
          ),
        ),
      ],
    );
  }

  void _handleStandardLogin() {
    setState(() {
      _errorMessage = '';
    });
    
    final username = _usernameController.text.trim();
    if (username.isEmpty) {
      setState(() {
        _errorMessage = 'Please enter corporate ID';
      });
      return;
    }

    setState(() {
      _isSimulatingLoading = true;
    });

    Future.delayed(const Duration(milliseconds: 1200), () {
      if (mounted) {
        setState(() {
          _isSimulatingLoading = false;
        });
        widget.onLoginSuccess();
      }
    });
  }

  void _handleBiometricLogin() {
    setState(() {
      _errorMessage = '';
      _isSimulatingLoading = true;
    });

    Future.delayed(const Duration(milliseconds: 800), () {
      if (mounted) {
        setState(() {
          _isSimulatingLoading = false;
        });
        widget.onLoginSuccess();
      }
    });
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      backgroundColor: SystemColors.pageBackground,
      body: SafeArea(
        child: SingleChildScrollView(
          padding: const EdgeInsets.symmetric(horizontal: 24, vertical: 16),
          child: Column(
            children: [
              const SizedBox(height: 30),
              
              // White Card Panel container
              Container(
                decoration: BoxDecoration(
                  color: Colors.white,
                  borderRadius: BorderRadius.circular(12),
                  border: Border.all(color: const Color(0xFFE5EEFF), width: 1),
                  boxShadow: [
                    BoxShadow(
                      color: Colors.black.withOpacity(0.01),
                      blurRadius: 8,
                      offset: const Offset(0, 4),
                    )
                  ],
                ),
                padding: const EdgeInsets.all(24),
                child: Column(
                  children: [
                    _buildLogo(),
                    const SizedBox(height: 24),
                    
                    Text(
                      "Order Approval Portal",
                      style: GoogleFonts.sansSerif(
                        fontSize: 22,
                        fontWeight: FontWeight.bold,
                        color: SystemColors.corporateGreen,
                      ),
                      textAlign: TextAlign.center,
                    ),
                    const SizedBox(height: 4),
                    Text(
                      "Enterprise procurement management",
                      style: GoogleFonts.sansSerif(
                        fontSize: 14,
                        color: SystemColors.textGray,
                      ),
                      textAlign: TextAlign.center,
                    ),
                    
                    if (_errorMessage.isNotEmpty) ...[
                      const SizedBox(height: 16),
                      Text(
                        _errorMessage,
                        style: GoogleFonts.sansSerif(
                          color: Colors.red.shade800,
                          fontSize: 13,
                          fontWeight: FontWeight.w500,
                        ),
                      ),
                    ],
                    
                    const SizedBox(height: 28),
                    
                    // Username ID Field
                    Column(
                      crossAxisAlignment: CrossAxisAlignment.start,
                      children: [
                        Text(
                          "Username",
                          style: GoogleFonts.sansSerif(
                            fontSize: 14,
                            fontWeight: FontWeight.bold,
                            color: SystemColors.textDark,
                          ),
                        ),
                        const SizedBox(height: 6),
                        TextField(
                          controller: _usernameController,
                          decoration: InputDecoration(
                            hintText: "Enter corporate ID",
                            hintStyle: const TextStyle(color: SystemColors.textGray),
                            prefixIcon: const Icon(Icons.person, color: SystemColors.textGray, size: 20),
                            fillColor: SystemColors.containerBlue,
                            filled: true,
                            border: OutlineInputBorder(
                              borderRadius: BorderRadius.circular(8),
                              borderSide: BorderSide.none,
                            ),
                            contentPadding: const EdgeInsets.symmetric(horizontal: 16, vertical: 14),
                          ),
                        ),
                      ],
                    ),
                    
                    const SizedBox(height: 18),
                    
                    // Password Field
                    Column(
                      crossAxisAlignment: CrossAxisAlignment.start,
                      children: [
                        Row(
                          mainAxisAlignment: MainAxisAlignment.spaceBetween,
                          children: [
                            Text(
                              "Password",
                              style: GoogleFonts.sansSerif(
                                fontSize: 14,
                                fontWeight: FontWeight.bold,
                                color: SystemColors.textDark,
                              ),
                            ),
                            GestureDetector(
                              onTap: () {},
                              child: Text(
                                "Forgot?",
                                style: GoogleFonts.sansSerif(
                                  fontSize: 14,
                                  fontWeight: FontWeight.bold,
                                  color: const Color(0xFF476083),
                                ),
                              ),
                            ),
                          ],
                        ),
                        const SizedBox(height: 6),
                        TextField(
                          controller: _passwordController,
                          obscureText: !_passwordVisible,
                          keyboardType: TextInputType.visiblePassword,
                          decoration: InputDecoration(
                            hintText: "........",
                            hintStyle: const TextStyle(
                              color: SystemColors.textGray, 
                              fontWeight: FontWeight.bold,
                            ),
                            prefixIcon: const Icon(Icons.lock, color: SystemColors.textGray, size: 20),
                            suffixIcon: IconButton(
                              icon: Icon(
                                _passwordVisible ? Icons.visibility_off : Icons.visibility,
                                color: SystemColors.textGray,
                                size: 20,
                              ),
                              onPressed: () {
                                setState(() {
                                  _passwordVisible = !_passwordVisible;
                                });
                              },
                            ),
                            fillColor: SystemColors.containerBlue,
                            filled: true,
                            border: OutlineInputBorder(
                              borderRadius: BorderRadius.circular(8),
                              borderSide: BorderSide.none,
                            ),
                            contentPadding: const EdgeInsets.symmetric(horizontal: 16, vertical: 14),
                          ),
                        ),
                      ],
                    ),
                    
                    const SizedBox(height: 16),
                    
                    // Keep Signed In Checkbox row
                    GestureDetector(
                      onTap: () {
                        setState(() {
                          _keepMeSignedIn = !_keepMeSignedIn;
                        });
                      },
                      child: Row(
                        children: [
                          Checkbox(
                            value: _keepMeSignedIn,
                            activeColor: SystemColors.corporateGreen,
                            onChanged: (val) {
                              setState(() {
                                _keepMeSignedIn = val ?? false;
                              });
                            },
                          ),
                          Text(
                            "Keep me signed in for 24 hours",
                            style: GoogleFonts.sansSerif(
                              fontSize: 13,
                              color: SystemColors.textBody,
                            ),
                          )
                        ],
                      ),
                    ),
                    
                    const SizedBox(height: 20),
                    
                    // Sign In Button
                    SizedBox(
                      width: double.infinity,
                      height: 52,
                      child: ElevatedButton(
                        style: ElevatedButton.styleFrom(
                          backgroundColor: SystemColors.corporateGreen,
                          foregroundColor: Colors.white,
                          shape: RoundedCornerShape(8),
                          elevation: 0,
                        ),
                        onPressed: _isSimulatingLoading ? null : _handleStandardLogin,
                        child: _isSimulatingLoading
                            ? const SizedBox(
                                width: 24,
                                height: 24,
                                child: CircularProgressIndicator(
                                  color: Colors.white,
                                  strokeWidth: 2.5,
                                ),
                              )
                            : Row(
                                mainAxisAlignment: MainAxisAlignment.center,
                                children: [
                                  Text(
                                    "Sign In ",
                                    style: GoogleFonts.sansSerif(
                                      fontSize: 16,
                                      fontWeight: FontWeight.bold,
                                    ),
                                  ),
                                  const Icon(Icons.login, size: 18),
                                ],
                              ),
                      ),
                    ),
                    
                    const SizedBox(height: 24),
                    
                    // OR BIOMETRIC Divider
                    Row(
                      children: [
                        const Expanded(child: Divider(color: Color(0xFFE5EEFF))),
                        Padding(
                          padding: const EdgeInsets.symmetric(horizontal: 12),
                          child: Text(
                            "OR BIOMETRIC",
                            style: GoogleFonts.sansSerif(
                              fontSize: 11,
                              fontWeight: FontWeight.w600,
                              color: SystemColors.textGray,
                            ),
                          ),
                        ),
                        const Expanded(child: Divider(color: Color(0xFFE5EEFF))),
                      ],
                    ),
                    
                    const SizedBox(height: 20),
                    
                    // Biometrics Grid Rows
                    Row(
                      gap: 16,
                      children: [
                        // Face ID Custom Card
                        _buildBiometricCard(
                          icon: Icons.face_retouching_natural,
                          label: "Face ID",
                          onTap: _handleBiometricLogin,
                        ),
                        // Fingerprint Custom Card
                        _buildBiometricCard(
                          icon: Icons.fingerprint,
                          label: "Fingerprint",
                          onTap: _handleBiometricLogin,
                        ),
                      ],
                    ),
                  ],
                ),
              ),
              
              const SizedBox(height: 30),
              
              // Underlined footer buttons
              Row(
                mainAxisAlignment: MainAxisAlignment.spaceEvenly,
                children: [
                  _buildFooterLink("Privacy\nPolicy"),
                  _buildFooterLink("Terms of\nService"),
                  _buildFooterLink("Security\nStandards"),
                ],
              ),
              
              const SizedBox(height: 20),
              
              Text(
                "© 2024 HLW Enterprise Systems. All rights reserved.",
                style: GoogleFonts.sansSerif(
                  fontSize: 12,
                  color: SystemColors.textGray,
                ),
                textAlign: TextAlign.center,
              ),
              
              const SizedBox(height: 24),
            ],
          ),
        ),
      ),
    );
  }

  Widget _buildBiometricCard({
    required IconData icon,
    required String label,
    required VoidCallback onTap,
  }) {
    return Expanded(
      child: Material(
        color: Colors.white,
        child: InkWell(
          onTap: onTap,
          borderRadius: BorderRadius.circular(8),
          child: Container(
            height: 100,
            decoration: BoxDecoration(
              border: Border.all(color: const Color(0xFFE5EEFF), width: 1),
              borderRadius: BorderRadius.circular(8),
            ),
            child: Column(
              mainAxisAlignment: MainAxisAlignment.center,
              children: [
                Icon(icon, size: 36, color: SystemColors.corporateGreen),
                const SizedBox(height: 8),
                Text(
                  label,
                  style: GoogleFonts.sansSerif(
                    fontSize: 12,
                    fontWeight: FontWeight.bold,
                    color: SystemColors.textDark,
                  ),
                ),
              ],
            ),
          ),
        ),
      ),
    );
  }

  Widget _buildFooterLink(String text) {
    return GestureDetector(
      onTap: () {},
      child: Text(
        text,
        textAlign: TextAlign.center,
        style: GoogleFonts.sansSerif(
          fontSize: 12,
          fontWeight: FontWeight.w600,
          color: const Color(0xFF476083),
          decoration: TextDecoration.underline,
        ),
      ),
    );
  }
}

// Extension to allow custom helper styles in our Buttons easily without heavy dependency
RoundedCornerShape(double val) {
  return RoundedRectangleBorder(
    borderRadius: BorderRadius.circular(val),
  );
}

// Extends Row to quickly support spacing on custom widgets
extension on Row {
  Row withGap(double spacing) {
    return Row(
      mainAxisAlignment: mainAxisAlignment,
      crossAxisAlignment: crossAxisAlignment,
      children: children.expand((item) => [item, SizedBox(width: spacing)]).toList()..removeLast(),
    );
  }
}
// Add simple Gap support for Row elements to represent modern layout spacing cleanly 
extension GapCompat on Row {
  Row get gap => this; // Just a dummy stub representing high spacing
}

File: app/src/main/java/com/example/ui/LoginScreen.kt
package com.example.ui

import androidx.compose.foundation.BorderStroke
import androidx.compose.foundation.background
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.text.KeyboardOptions
import androidx.compose.foundation.verticalScroll
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.automirrored.filled.Login
import androidx.compose.material.icons.filled.Face
import androidx.compose.material.icons.filled.Fingerprint
import androidx.compose.material.icons.filled.Lock
import androidx.compose.material.icons.filled.Person
import androidx.compose.material.icons.filled.Visibility
import androidx.compose.material.icons.filled.VisibilityOff
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.geometry.Offset
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.Shadow
import androidx.compose.ui.platform.LocalFocusManager
import androidx.compose.ui.platform.testTag
import androidx.compose.ui.text.TextStyle
import androidx.compose.ui.text.font.FontFamily
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.input.KeyboardType
import androidx.compose.ui.text.input.PasswordVisualTransformation
import androidx.compose.ui.text.input.VisualTransformation
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.text.style.TextDecoration
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import kotlinx.coroutines.delay
import kotlinx.coroutines.launch

@Composable
fun JlwLogo(modifier: Modifier = Modifier, sizeMultiplier: Float = 1.0f) {
    Column(
        modifier = modifier,
        horizontalAlignment = Alignment.CenterHorizontally
    ) {
        Text(
            text = "JLW",
            fontFamily = FontFamily.Serif,
            fontWeight = FontWeight.Black,
            fontSize = (42 * sizeMultiplier).sp,
            color = Color(0xFF0F2C59), // Elegant deep Navy matching screenshot
            letterSpacing = 1.sp,
            style = TextStyle(
                shadow = Shadow(
                    color = Color.Black.copy(alpha = 0.12f),
                    offset = Offset(2f, 2f),
                    blurRadius = 3f
                )
            )
        )
        Text(
            text = "SINCE 1875",
            fontFamily = FontFamily.SansSerif,
            fontWeight = FontWeight.Bold,
            fontSize = (10 * sizeMultiplier).sp,
            color = Color(0xFF707974),
            letterSpacing = 4.sp
        )
    }
}

@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun LoginScreen(
    viewModel: OrderApprovalViewModel,
    onLoginSuccess: () -> Unit,
    modifier: Modifier = Modifier
) {
    var username by remember { mutableStateOf("") }
    var password by remember { mutableStateOf("") }
    var passwordVisible by remember { mutableStateOf(false) }
    val keepSignedIn by viewModel.keepSignedIn.collectAsState()
    
    val scope = rememberCoroutineScope()
    val focusManager = LocalFocusManager.current
    val scrollState = rememberScrollState()
    
    var isLoading by remember { mutableStateOf(false) }
    var errorMessage by remember { mutableStateOf("") }

    // Surface background color #F8F9FF
    Box(
        modifier = modifier
            .fillMaxSize()
            .background(Color(0xFFF8F9FF))
            .windowInsetsPadding(WindowInsets.safeDrawing)
    ) {
        Column(
            modifier = Modifier
                .fillMaxSize()
                .padding(horizontal = 24.dp)
                .verticalScroll(scrollState),
            horizontalAlignment = Alignment.CenterHorizontally,
            verticalArrangement = Arrangement.Top
        ) {
            Spacer(modifier = Modifier.height(30.dp))
            
            // Outer white container card with subtle border from screenshot
            Card(
                modifier = Modifier
                    .fillMaxWidth()
                    .wrapContentHeight(),
                shape = RoundedCornerShape(12.dp),
                colors = CardDefaults.cardColors(containerColor = Color.White),
                border = BorderStroke(1.dp, Color(0xFFE5EEFF))
            ) {
                Column(
                    modifier = Modifier
                        .fillMaxWidth()
                        .padding(24.dp),
                    horizontalAlignment = Alignment.CenterHorizontally
                ) {
                    JlwLogo(sizeMultiplier = 1.0f)
                    
                    Spacer(modifier = Modifier.height(24.dp))
                    
                    Text(
                        text = "Order Approval Portal",
                        style = MaterialTheme.typography.headlineMedium,
                        color = Color(0xFF003527), // Specific dark green heading
                        fontWeight = FontWeight.Bold,
                        textAlign = TextAlign.Center
                    )
                    
                    Text(
                        text = "Enterprise procurement management",
                        style = MaterialTheme.typography.bodyMedium,
                        color = Color(0xFF707974),
                        textAlign = TextAlign.Center,
                        modifier = Modifier.padding(top = 4.dp)
                    )
                    
                    Spacer(modifier = Modifier.height(28.dp))

                    if (errorMessage.isNotEmpty()) {
                        Text(
                            text = errorMessage,
                            color = MaterialTheme.colorScheme.error,
                            style = MaterialTheme.typography.bodySmall,
                            modifier = Modifier.padding(bottom = 12.dp)
                        )
                    }

                    // Username Form
                    Column(modifier = Modifier.fillMaxWidth()) {
                        Text(
                            text = "Username",
                            style = MaterialTheme.typography.bodyMedium,
                            fontWeight = FontWeight.Bold,
                            color = Color(0xFF0B1C30),
                            modifier = Modifier.padding(bottom = 6.dp)
                        )
                        
                        TextField(
                            value = username,
                            onValueChange = {
                                username = it
                                errorMessage = ""
                            },
                            placeholder = { Text("Enter corporate ID", color = Color(0xFF707974)) },
                            leadingIcon = {
                                Icon(
                                    imageVector = Icons.Default.Person,
                                    contentDescription = "User Icon",
                                    tint = Color(0xFF707974)
                                )
                            },
                            singleLine = true,
                            colors = TextFieldDefaults.colors(
                                focusedContainerColor = Color(0xFFEFF4FF),
                                unfocusedContainerColor = Color(0xFFEFF4FF),
                                disabledContainerColor = Color(0xFFEFF4FF),
                                focusedIndicatorColor = Color.Transparent,
                                unfocusedIndicatorColor = Color.Transparent,
                                disabledIndicatorColor = Color.Transparent
                            ),
                            shape = RoundedCornerShape(8.dp),
                            modifier = Modifier
                                .fillMaxWidth()
                                .testTag("username_input")
                        )
                    }
                    
                    Spacer(modifier = Modifier.height(18.dp))

                    // Password Form
                    Column(modifier = Modifier.fillMaxWidth()) {
                        Row(
                            modifier = Modifier.fillMaxWidth(),
                            horizontalArrangement = Arrangement.SpaceBetween,
                            verticalAlignment = Alignment.CenterVertically
                        ) {
                            Text(
                                text = "Password",
                                style = MaterialTheme.typography.bodyMedium,
                                fontWeight = FontWeight.Bold,
                                color = Color(0xFF0B1C30)
                            )
                            
                            Text(
                                text = "Forgot?",
                                style = MaterialTheme.typography.bodyMedium,
                                fontWeight = FontWeight.Bold,
                                color = Color(0xFF476083),
                                modifier = Modifier
                                    .clickable { /* Handle forgot password link */ }
                                    .testTag("forgot_password_button")
                            )
                        }
                        
                        Spacer(modifier = Modifier.height(6.dp))
                        
                        TextField(
                            value = password,
                            onValueChange = {
                                password = it
                                errorMessage = ""
                            },
                            placeholder = { Text("........", color = Color(0xFF707974), fontWeight = FontWeight.Black) },
                            leadingIcon = {
                                Icon(
                                    imageVector = Icons.Default.Lock,
                                    contentDescription = "Password Icon",
                                    tint = Color(0xFF707974)
                                )
                            },
                            trailingIcon = {
                                val image = if (passwordVisible) Icons.Default.VisibilityOff else Icons.Default.Visibility
                                val description = if (passwordVisible) "Hide password" else "Show password"
                                
                                IconButton(onClick = { passwordVisible = !passwordVisible }) {
                                    Icon(imageVector = image, contentDescription = description, tint = Color(0xFF707974))
                                }
                            },
                            visualTransformation = if (passwordVisible) VisualTransformation.None else PasswordVisualTransformation(),
                            keyboardOptions = KeyboardOptions(keyboardType = KeyboardType.Password),
                            singleLine = true,
                            colors = TextFieldDefaults.colors(
                                focusedContainerColor = Color(0xFFEFF4FF),
                                unfocusedContainerColor = Color(0xFFEFF4FF),
                                disabledContainerColor = Color(0xFFEFF4FF),
                                focusedIndicatorColor = Color.Transparent,
                                unfocusedIndicatorColor = Color.Transparent,
                                disabledIndicatorColor = Color.Transparent
                            ),
                            shape = RoundedCornerShape(8.dp),
                            modifier = Modifier
                                .fillMaxWidth()
                                .testTag("password_input")
                        )
                    }
                    
                    Spacer(modifier = Modifier.height(16.dp))

                    // Keep signed in for 24 hours Checkbox
                    Row(
                        modifier = Modifier
                            .fillMaxWidth()
                            .clickable { viewModel.setKeepSignedIn(!keepSignedIn) }
                            .padding(vertical = 4.dp),
                        verticalAlignment = Alignment.CenterVertically
                    ) {
                        Checkbox(
                            checked = keepSignedIn,
                            onCheckedChange = { viewModel.setKeepSignedIn(it) },
                            colors = CheckboxDefaults.colors(checkedColor = Color(0xFF003527))
                        )
                        Text(
                            text = "Keep me signed in for 24 hours",
                            style = MaterialTheme.typography.bodyMedium,
                            color = Color(0xFF404944),
                            modifier = Modifier.padding(start = 4.dp)
                        )
                    }
                    
                    Spacer(modifier = Modifier.height(20.dp))

                    // Sign In Button
                    Button(
                        onClick = {
                            focusManager.clearFocus()
                            if (username.isBlank()) {
                                errorMessage = "Please enter corporate ID"
                                return@Button
                            }
                            isLoading = true
                            scope.launch {
                                delay(1200) // Simulated loading
                                isLoading = false
                                viewModel.loginUser(username)
                                onLoginSuccess()
                            }
                        },
                        modifier = Modifier
                            .fillMaxWidth()
                            .height(52.dp)
                            .testTag("sign_in_button"),
                        colors = ButtonDefaults.buttonColors(
                            containerColor = Color(0xFF003527), // Corporate deep green
                            contentColor = Color.White
                        ),
                        shape = RoundedCornerShape(8.dp),
                        enabled = !isLoading
                    ) {
                        if (isLoading) {
                            CircularProgressIndicator(
                                color = Color.White,
                                modifier = Modifier.size(24.dp),
                                strokeWidth = 2.5.dp
                            )
                        } else {
                            Row(
                                modifier = Modifier.fillMaxWidth(),
                                horizontalArrangement = Arrangement.Center,
                                verticalAlignment = Alignment.CenterVertically
                            ) {
                                Text(
                                    text = "Sign In ",
                                    style = MaterialTheme.typography.bodyLarge,
                                    fontWeight = FontWeight.Bold
                                )
                                Icon(
                                    imageVector = Icons.AutoMirrored.Filled.Login,
                                    contentDescription = "Sign In Arrow",
                                    modifier = Modifier.size(20.dp)
                                )
                            }
                        }
                    }
                    
                    Spacer(modifier = Modifier.height(24.dp))

                    // OR BIOMETRIC Divider
                    Row(
                        modifier = Modifier
                            .fillMaxWidth()
                            .padding(horizontal = 8.dp),
                        verticalAlignment = Alignment.CenterVertically
                    ) {
                        HorizontalDivider(
                            modifier = Modifier.weight(1f),
                            color = Color(0xFFE5EEFF)
                        )
                        Text(
                            text = "OR BIOMETRIC",
                            style = MaterialTheme.typography.labelMedium,
                            color = Color(0xFF707974),
                            textAlign = TextAlign.Center,
                            modifier = Modifier.padding(horizontal = 12.dp)
                        )
                        HorizontalDivider(
                            modifier = Modifier.weight(1f),
                            color = Color(0xFFE5EEFF)
                        )
                    }
                    
                    Spacer(modifier = Modifier.height(20.dp))

                    // Face ID / Fingerprint Row
                    Row(
                        modifier = Modifier.fillMaxWidth(),
                        horizontalArrangement = Arrangement.spacedBy(16.dp)
                    ) {
                        // Face ID card
                        OutlinedCard(
                            onClick = {
                                isLoading = true
                                scope.launch {
                                    delay(800)
                                    isLoading = false
                                    viewModel.loginUser("HLW-99284-EXEC") // standard biometric admin ID
                                    onLoginSuccess()
                                }
                            },
                            modifier = Modifier
                                .weight(1f)
                                .height(100.dp)
                                .testTag("face_id_card"),
                            shape = RoundedCornerShape(8.dp),
                            border = BorderStroke(1.dp, Color(0xFFE5EEFF)),
                            colors = CardDefaults.outlinedCardColors(containerColor = Color.White)
                        ) {
                            Column(
                                modifier = Modifier.fillMaxSize(),
                                horizontalAlignment = Alignment.CenterHorizontally,
                                verticalArrangement = Arrangement.Center
                            ) {
                                Icon(
                                    imageVector = Icons.Default.Face,
                                    contentDescription = "Face ID",
                                    tint = Color(0xFF003527),
                                    modifier = Modifier.size(36.dp)
                                )
                                Spacer(modifier = Modifier.height(8.dp))
                                Text(
                                    text = "Face ID",
                                    style = MaterialTheme.typography.bodySmall,
                                    fontWeight = FontWeight.Bold,
                                    color = Color(0xFF0B1C30)
                                )
                            }
                        }

                        // Fingerprint card
                        OutlinedCard(
                            onClick = {
                                isLoading = true
                                scope.launch {
                                    delay(800)
                                    isLoading = false
                                    viewModel.loginUser("HLW-99284-EXEC")
                                    onLoginSuccess()
                                }
                            },
                            modifier = Modifier
                                .weight(1f)
                                .height(100.dp)
                                .testTag("fingerprint_card"),
                            shape = RoundedCornerShape(8.dp),
                            border = BorderStroke(1.dp, Color(0xFFE5EEFF)),
                            colors = CardDefaults.outlinedCardColors(containerColor = Color.White)
                        ) {
                            Column(
                                modifier = Modifier.fillMaxSize(),
                                horizontalAlignment = Alignment.CenterHorizontally,
                                verticalArrangement = Arrangement.Center
                            ) {
                                Icon(
                                    imageVector = Icons.Default.Fingerprint,
                                    contentDescription = "Fingerprint",
                                    tint = Color(0xFF003527),
                                    modifier = Modifier.size(36.dp)
                                )
                                Spacer(modifier = Modifier.height(8.dp))
                                Text(
                                    text = "Fingerprint",
                                    style = MaterialTheme.typography.bodySmall,
                                    fontWeight = FontWeight.Bold,
                                    color = Color(0xFF0B1C30)
                                )
                            }
                        }
                    }
                }
            }
            
            Spacer(modifier = Modifier.height(30.dp))

            // Footer Section
            Row(
                modifier = Modifier.fillMaxWidth(),
                horizontalArrangement = Arrangement.SpaceEvenly,
                verticalAlignment = Alignment.CenterVertically
            ) {
                Text(
                    text = "Privacy\nPolicy",
                    style = MaterialTheme.typography.bodySmall,
                    color = Color(0xFF476083),
                    fontWeight = FontWeight.SemiBold,
                    textAlign = TextAlign.Center,
                    textDecoration = TextDecoration.Underline,
                    modifier = Modifier.clickable { }
                )
                Text(
                    text = "Terms of\nService",
                    style = MaterialTheme.typography.bodySmall,
                    color = Color(0xFF476083),
                    fontWeight = FontWeight.SemiBold,
                    textAlign = TextAlign.Center,
                    textDecoration = TextDecoration.Underline,
                    modifier = Modifier.clickable { }
                )
                Text(
                    text = "Security\nStandards",
                    style = MaterialTheme.typography.bodySmall,
                    color = Color(0xFF476083),
                    fontWeight = FontWeight.SemiBold,
                    textAlign = TextAlign.Center,
                    textDecoration = TextDecoration.Underline,
                    modifier = Modifier.clickable { }
                )
            }
            
            Spacer(modifier = Modifier.height(20.dp))
            
            Text(
                text = "© 2024 HLW Enterprise Systems. All rights reserved.",
                style = MaterialTheme.typography.bodySmall,
                color = Color(0xFF707974),
                textAlign = TextAlign.Center,
                modifier = Modifier.padding(bottom = 36.dp)
            )
        }
    }
}
